public/typescript/operators.ts

```diff
@@ -10,3 +10,9 @@ export const requestUserAssertion = concatMap((publicKey: PublicKeyCredentialCre
 export const filterPublicKeyCredentialType = filter(isPublicKeyCredentialType);
 
 export const excludePublicKeyCredentialType = filter(complement(isPublicKeyCredentialType));
+
+export const concatIfElse = <T, R1, R2>(
```

Comment:
Maybe you could rename this function to make it expressive in the context of the code that uses it? Maybe renaming it to something like 'when' making it read

`when isPublicKeyCredentialType`